packages/plugins/typescript/typed-document-node/tests/typed-document-node.spec.ts

```diff
@@ -7,4 +8,54 @@ describe('TypeDocumentNode', () => {
     expect(result.content).toBe('');
     expect(result.prepend.length).toBe(0);
   });
+
+  it('Duplicated nested fragments handle', async () => {
+    const schema = buildSchema(/* GraphQL */ `
+      schema {
+        query: Query
+      }
+
+      type Query {
+        jobs: [Job!]!
+      }
+
+      type Job {
+        id: ID!
+        recruiterName: String!
+        title: String!
+      }
+    `);
+
+    const ast = parse(/* GraphQL */ `
+      query GetJobs {
```

Comment:
Please add an additional test where the fragment is loaded also by the query (like this one, but we `...JobSimpleRecruiterData` as well)